jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -417,3 +417,46 @@ object XGBoost extends Serializable {
     }
   }
 }
+
+private class Watches private(val train: DMatrix, val test: DMatrix) {
```

Comment:
what if the user wants to monitor more than 2 datasets?

Reply:
This is somewhat tricky to do using spark.ml APIs, so I planned to delay the implementation until there's a request. Otherwise, we could change `Watches` to wrap a `Map` and expose train/test as properties.